### PR TITLE
feat: UI and UX improvements for Table of Contents

### LINKED ISSUE #864: make the Table of Contents component sticky
**Please describe the feature**

As a user, I want the Table of Contents to stick to the top so that it does not disappear as I scroll down the web page.

**Describe the solution you would like**

Make sure that the Table of Contents (currently for the Documentation and About pages) is always visible even if the user scrolls to the bottom of a long page that surpasses the height of Table of Contents.

**Additional context**

Two things to keep in mind for implementation:
- The mobile layout is different (ToC at top, content at bottom) and it should not be sticky depending on how things are laid out.
- If the ToC is longer than the visible window height, it should be possible to reach the bottom of the ToC as well.
- For the active link, nested visualization should be added similarly to example below.

image

### PR DESCRIPTION
**Pull request by e0**
**Related issue(s) and PR(s)**  
This PR closes #864.

**Type of change**  

- [x] Other
 
**List of changes made**  

Changes for the ToC are made according to the issue description, including:
- Sticks to the top after scrolling the page down.
- Has max visible height of screen height and gets its own scroll section if it's longer than the screen height.
- Does not become sticky for mobile (smaller than `$tablet`) devices.
- Adds nested visualization for active link. The active color before `white-ter` is now used for the active parent link. The actual active link color is now `link-light`, which seems to work well with the look and feel of the rest of the ToC, for example the icon color.

**Screenshot of the fix**  

image
image
image
image

**Testing**  

- Relative urls that can be reused both for production and local testing
- Instructions on how to test

Try clicking the ToC links as well as scrolling the `/documentation` and `/about/*` pages.

**Further comments**  

- Please let me know if you have any suggestions on the design choices made in this PR.
- I have tested using different screen sizes on Chrome, Firefox, and Safari on macOS.

**Definition of Done checklist**  
- [x] My code meets the Acceptance Criteria
- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [x] Tests and lint/format validations are passing
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by inghylt** [comment]
Really nice! I've tried it out and messed with screen sizes on firefox and chrome, worked well. Minor thought, not sure how common/pleasant it is to scroll the left side index(ToC) as you need at `/documentation`. Maybe we could decrease the space between each sub item in the ToC to make it fit and remove the need for scrolling?

**Comment by e0** [comment]
I’m open to decreasing the spacing but just to clarify it won’t guarantee not needing to scroll since that depends on the user’s window height. I think this technique is quite common in documentation pages and also in our map viewer sidebar. They do usually have clearer visual separation from the rest of the page though and I think there is room for improvement there as well. A new issue could be created if we want to pursue it.

File: frontend/src/components/shared/TableOfContents.vue [hunk 1]
```diff
@@ -39,10 +40,25 @@ export default {
...
+  },
 };
 </script>
 
 <style lang="scss" scoped>
+@media screen and (min-width: $tablet) {
+  #table-of-contents {
+    position: sticky;
+    top: 0;
+    align-self: flex-start;
```
Comment (inghylt) [inline frontend/src/components/shared/TableOfContents.vue:56]: Is it the Bulma class `container` in `./frontend/src/layouts/AboutLayout.vue` and `./frontend/src/components/Documentation.vue` that makes this div a child to a flexible container?

**Review by inghylt — APPROVED** [review]
Great job!

File: frontend/src/components/shared/TableOfContents.vue (hunk 1 above)
Comment (e0) [reply frontend/src/components/shared/TableOfContents.vue:56]: Good question! I think it's the `columns` class that adds the `display: flex;` style. Really nice that you spotted this as it points to an opportunity to further refactor these similar layouts.

**Review by MalinAhlberg — APPROVED** [review]
Looks really nice, great work!

**Comment by mihai-sysbio** [comment]
Looks neat! I'm partly having second thoughts when I'm seeing that vertical scrollbar appear for the ToC, but I guess it will be a downside only for a small fraction of users, whereas it will be a improvement for most 👍🏻 

Regarding the use of the new class `has-background-link-light` - I think this is a new color that we haven't used anywhere on the website, so my immediate reactions are:
- is there any existing class that could be used?
- personally I find the blue too intense: the other colors in the palette defined in `vars.css` are more pastel, especially when it comes to a wider area (surface) taken by the visual element (a highlight is much larger than an icon)

**Comment by e0** [comment]
> Looks neat! I'm partly having second thoughts when I'm seeing that vertical scrollbar appear for the ToC, but I guess it will be a downside only for a small fraction of users, whereas it will be a improvement for most 👍🏻

Yea it's not ideal with the scrollbar and it's a trade off we have to settle with for now. In the future we could consider improving how it looks as mentioned [here](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/920#issuecomment-1070736463).

___

> Regarding the use of the new class `has-background-link-light` - I think this is a new color that we haven't used anywhere on the website, so my immediate reactions are:
> 
> * is there any existing class that could be used?
> * personally I find the blue too intense: the other colors in the palette defined in `vars.css` are more pastel, especially when it comes to a wider area (surface) taken by the visual element (a highlight is much larger than an icon)

The choice of a blueish color is so that it should play well with our existing color for the ToC icons, which currently has `has-text-info`. In the `vars.scss` we have actually defined `$link: #00549e;` and `$info: $link;`. I'm open to suggestions if you have any colors to recommend. Not sure exactly which adjustments to make since I actually think this shade of blue is more pastelly than the other colors in `vars.scss`.

**Review by nanjiangshu — APPROVED** [review]
Really nice implementation! I tested on different Browsers with different dimensions and it all worked as expected. 
There might still be room for improvement for instance when only part of the ToC is shown (with a larger font), it is not obvious that there are more content existing in the ToC when the scroll bar is hidden. Nevertheless, I think the current solution is good enough for this issue.

**Comment by e0** [comment]
> There might still be room for improvement for instance when only part of the ToC is shown (with a larger font), it is not obvious that there are more content existing in the ToC when the scroll bar is hidden. Nevertheless, I think the current solution is good enough for this issue.

That's a good point @nanjiangshu. I think we could also consider having the option to collapse and expand the menu.

**Pull request merged**